exts/cesium.omniverse/mdl/cesium.mdl

```diff
@@ -1602,3 +1607,54 @@ export gltf_texture_lookup_value cesium_internal_imagery_layer_resolver(
 
     return gltf_texture_lookup_value(true, resolved_value);
 }
+
+export gltf_texture_lookup_value cesium_internal_polygon_imagery_layer_resolver(
+    uniform int polygon_imagery_layers_count = 0,
+    gltf_texture_lookup_value imagery_layer_0 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_1 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_2 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_3 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_4 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_5 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_6 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_7 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_8 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_9 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_10 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_11 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_12 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_13 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_14 = gltf_texture_lookup_value(),
+    gltf_texture_lookup_value imagery_layer_15 = gltf_texture_lookup_value()
+) [[ anno::hidden() ]] {
+    // The array length should match MAX_IMAGERY_LAYERS_COUNT in FabricMaterial.cpp
+    gltf_texture_lookup_value[] imagery_layers(
+        imagery_layer_0,
+        imagery_layer_1,
+        imagery_layer_2,
+        imagery_layer_3,
+        imagery_layer_4,
+        imagery_layer_5,
+        imagery_layer_6,
+        imagery_layer_7,
+        imagery_layer_8,
+        imagery_layer_9,
+        imagery_layer_10,
+        imagery_layer_11,
+        imagery_layer_12,
+        imagery_layer_13,
+        imagery_layer_14,
+        imagery_layer_15,
+    );
+
+    auto resolved_value = float4(0.0);
+
+    for (int i = 0; i < polygon_imagery_layers_count; i++) {
+        auto imagery_layer = imagery_layers[i];
+        if (imagery_layer.valid) {
+            resolved_value += imagery_layer.value;
+        }
+    }
```

Comment:
Yeah, we could do that. It's maybe better.

If you interpret what Native returns as a mask, then it's inverted: 1.0 is intended to clip and 0 not clip. It feels odd when clipping but seems more natural if stacking non-clip overlay polygons additively.

Reply:
The code above will in most cases produce only a black single-channel texture. It will only return 1.0 values if all the clipping polygons overlap at that pixel. I suppose we could flip the texture value and do `resolved_value *= (1.0 - imagery_layer.value);` So far every solution had a bit of awkwardness. Maybe we resolve this when dropping in colored polygon overlays?

Reply:
> If you interpret what Native returns as a mask, then it's inverted: 1.0 is intended to clip and 0 not clip. It feels odd when clipping but seems more natural if stacking non-clip overlay polygons additively.

Ah, if that's how native treats it then this approach is fine